### PR TITLE
Add basic framework for Casbin.js

### PR DESCRIPTION
**Pull request by kingiw**
I've created some basic api for casbin.js enforcer, which is used for syncing the user profiles.

1. `setUser` to set the current subject identity. When it is called, casbin.js enforcer will verify if the subject is modified and if there comes a new subject, it will try to sync with the backend server (call the `syncUserProfiles`)
2. `syncUserProfiles` to fetch subject profiles and save it at the frontend. 
3. `getProfiles` to get the user profiles.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece @GopherJ please review.

File: src/__test__/server.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    
+    public async run() {
+        // RBAC API doesn't support RBAC w/ domain.
+        // this.enforcer = await newEnforcer('./src/__test__/example/rbac_with_domains_model.conf', './src/__test__/example/rbac_with_domains_policy.csv');
+        this.enforcer = await newEnforcer('./src/__test__/example/rbac_model.conf', './src/__test__/example/rbac_policy.csv');
+    }
+    
+    public async getProfiles(sub: string) : Promise<string> {
+        const policies = await this.enforcer.getImplicitPermissionsForUser(sub);
+        let profiles : { [key:string]: string[] } = {};
```
Comment (nodece) [inline src/__test__/server.ts:16]: I think it's better to use `Map`.

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+interface BaseResponse {
+    message: string;
+    data: any;
+}
+
+export class Enforcer {
+    private endpoint: string;
+    private user : string | undefined;
+    private profiles: {[key: string]: string[]} = {};
```
Comment (nodece) [inline src/index.ts:11]: I think it's better to use `Map`.

File: src/__test__/server.ts (hunk 1 above)
Comment (kingiw) [reply src/__test__/server.ts:16]: Yep, I agree with that

**Comment by nodece** [comment]
@kingiw You also need to update package.json.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "casbin.js",
+  "version": "0.0.1",
+  "main": "`index.js",
+  "repository": "https://github.com/kingiw/casbin.js.git",
```
Comment (nodece) [inline package.json:5]: miss `typings`

File: src/__test__/server.ts (hunk 1 above)
Comment (kingiw) [reply src/__test__/server.ts:16]: After some consideration, I believe that use the raw object type `{ [key:string]: string[] }` may be more easy to manipulate the change on the objects' value. If we use Map<string, string[]> here, I think the code could be quite lengthy:
```typescript
let profiles : Map<string, string[]> = new Map<string, string[]>();
// Firstly, check if the key(action) exists
if (!profiles.has(action)) {
    profiles.set(action, []);
}
// Then take out the string array from the map, push new object inside, and set it back to the map
let arr : string[] = profiles.get(action);
arr.push(another_object);
profiles.set(action, arr);
// We need to pass the data to the frontend, so the map needs to be convert to JSON string
// However, apply JSON.stringify on a Map<string, string> instance doesn't work properly
// We need to convert Map<string, string> to object before using JSON.stringify
let obj : {[key: string]: string[]} = {};
profiles.forEach((value, key) => (obj[key] = value));
return JSON.stringify(obj);
```

In comparison, using `{ [key:string]: string[] }` here can be cleaner and more readable.

File: src/index.ts (hunk 2 above)
Comment (kingiw) [reply src/index.ts:11]: Use `Map` at the frontend is reasonable. Actually, I will implement a class `Profiles` for manipulate the data and relevant methods.

File: src/__test__/server.ts (hunk 1 above)
Comment (kingiw) [reply src/__test__/server.ts:16]: This is a simple implementation of generating user profiles at the backend. I wish to add this api to the Casbin/node-casbin library, which can ease the process of getting a profiles of a given user. The implementation details can be discussed later.

**Comment by kingiw** [comment]
> @kingiw You also need to update package.json such as author, repository and so on.

Who should I fill in the author?

File: src/__test__/server.ts (hunk 1 above)
Comment (nodece) [reply src/__test__/server.ts:16]: okay, keep the `{ [key:string]: string[] }`.

**Comment by nodece** [comment]
> > @kingiw You also need to update package.json such as author, repository and so on.
> 
> Who should I fill in the author?

You just need to update the `repository` field.

**Comment by nodece** [comment]
@hsluoyz waiting for you to review.

**Comment by kingiw** [comment]
@hsluoyz Plz review when available.

**Comment by hsluoyz** [comment]
lgtm

**Pull request merged**